### ISSUE TITLE
Python: Timestamp unit ignored on deserliazation

### ISSUE THREAD

**Issue by kevin-montrose** [wesm/feather#303]
The python library (version 0.3.1) doesn't respect the unit provided as metadata when deserializing, it appears to always assume `NANOSECOND`.  The R implementation behaves correctly.

I've attached a zip with the same date (1971-02-03 04:05:06 UTC) as the single value in a single column that demonstrates the issue.

Python:
![image](https://cloud.githubusercontent.com/assets/1219467/26315962/c780d21e-3ee0-11e7-9fa5-753bc7cdc5cc.png)

R:
![image](https://cloud.githubusercontent.com/assets/1219467/26316032/f6431ea4-3ee0-11e7-80fe-460a71a5cbcd.png)

[Timestamp.zip](https://github.com/wesm/feather/files/1019299/Timestamp.zip)

**Comment by wesm** [comment]
I think this is fixed in the move to Arrow. I am making a Python release soon that uses pyarrow as a dependency

**Comment by wesm** [comment]
Confirmed that this is fixed in Apache Arrow 0.4.0. I will make a Feather release for Python including this in the near future

```
In [7]: import pyarrow.feather as f

In [8]: f.read_feather('/home/wesm/Downloads/Timestamp_Seconds.feather')
Out[8]: 
                  Timestamp
0 1971-02-03 04:05:06+00:00

In [9]: f.read_feather('/home/wesm/Downloads/Timestamp_Milliseconds.feather')
Out[9]: 
                  Timestamp
0 1971-02-03 04:05:06+00:00

In [10]: f.read_feather('/home/wesm/Downloads/Timestamp_Microseconds.feather')
Out[10]: 
                  Timestamp
0 1971-02-03 04:05:06+00:00

In [11]: f.read_feather('/home/wesm/Downloads/Timestamp_Nanoseconds.feather')
Out[11]: 
                  Timestamp
0 1971-02-03 04:05:06+00:00
```
